autoload/cmake.vim

```diff
@@ -162,3 +163,34 @@ function! cmake#Close() abort
     call s:logger.LogDebug('API invoked: cmake#Close()')
     call s:terminal.Close()
 endfunction
+
+" API function for third-party plugins to query information
+"
+" Returns:
+"     Dictionary
+"         config : String
+"             name of the set cmake configuration
+"         status : String
+"             current cmake status (e.g. Building...)
+"         cmake_version : Dictionary
+"             major : Number
+"                 cmake major version
+"             minor : Number
+"                 cmake minor version
+"             patch : Number
+"                 cmake patch version
+"             string : String
+"                 cmake version in string representation
+"         project_dir : String
+"             absolute path to the vcs root
```

Comment:
Technically it doesn't have to be a vcs project, one could set different root markers and vim-cmake would still work. Let's reword the description a little

Reply:
How would you reword that then? I thought about this shortly before writing this but this seemed best. Something like 'absolute path to detected project root (see `g:cmake_root_markers`)' maybe?

Reply:
Yes that works, or even without the suggestion in parentheses